flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -789,4 +790,50 @@ public void configure(ReadableConfig configuration) {
                 .getOptional(CheckpointingOptions.CHECKPOINTS_DIRECTORY)
                 .ifPresent(this::setCheckpointStorage);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        CheckpointConfig that = (CheckpointConfig) o;
+        return checkpointInterval == that.checkpointInterval
```

Comment:
this seems fragile, when the new field will be added, this could not get updated. Could we do the comparison of the serialized form (original vs user) instead? I'd expect the serialized form to be stable within the same JVM.

Reply:
I changed to using serialization to check whether something has changed.